### PR TITLE
template: remove whitespace before template tags

### PR DESCRIPTION
**Pull request by milahu**

### CODE REVIEW

File: tgarchive/example/template.html [hunk 1]
```diff
@@ -39,163 +40,171 @@
 			<header class="header">
 				<div class="logo">
 					<a href="{{ config.site_url }}"><img src="static/logo.svg" alt="" /></a>
-					<p class="desc">
+					<div class="desc">
```
Comment (milahu) [inline tgarchive/example/template.html:43]: div is more flexible than p, as div allows nesting

File: tgarchive/example/template.html [hunk 2]
```diff
@@ -39,163 +40,171 @@
...
 						<a href="{{ config.telegram_url.format(id=config.group) }}" rel="noreferer nopener nofollow">@{{ config.group }}</a> /
```
Comment (milahu) [inline tgarchive/example/template.html:44]: can we remove the @ before config.group? its not used by telegram

File: tgarchive/example/template.html [hunk 3]
```diff
@@ -39,163 +40,171 @@
...
+							if m.user.tags %}{%
+							for t in m.user.tags %}
+								<span class="tags">{{ t }}</span>{%
+							endfor %}{%
+							endif %}
+							<span class="date">{{ m.date.strftime("%I:%M %p, %d %b %Y") }}</span>
 						</div>
+						<div class="text">{%
+							if m.type == "message" %}{{"\n"}}{{
+								nl2br(m.content | escape) | safe | urlize }}{%
```
Comment (milahu) [inline tgarchive/example/template.html:133]: ... so the text is unindented

```html
        <div class="text">
a
<br />b
<br />c
        </div>
```

File: tgarchive/example/template.html [hunk 4]
```diff
@@ -14,16 +15,16 @@
...
-	{% if config.publish_rss_feed %}
-		<link rel="alternate" type="application/rss+xml" title="RSS feed " href="index.xml" />
-		<link rel="alternate" type="application/atom+xml" title="Atom feed " href="index.atom" />
-	{% endif %}
+	<link rel="alternate" type="application/rss+xml" title="RSS feed " href="index.xml" />
+	<link rel="alternate" type="application/atom+xml" title="Atom feed " href="index.atom" />{%
+	endif %}
 
 	<link rel="preconnect" href="https://fonts.gstatic.com">
-	<link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;600&display=swap" rel="stylesheet">
```
Comment (milahu) [inline tgarchive/example/template.html:25]: can we remove links to google?

File: tgarchive/example/template.html [hunk 5]
```diff
@@ -14,16 +15,16 @@
 	<meta property="og:description" content="{{ meta_description }}" />
 	<meta property="og:type" content="website" />
 	<meta property="og:url" content="{{ config.site_url }}/.html" />
-	<meta property="og:image" content="{{ config.site_url }}/static/thumb.png" />
+	<meta property="og:image" content="{{ config.site_url }}/static/thumb.png" />{%
```
Comment (knadh) [inline tgarchive/example/template.html:18]: Joining lines to remove indentation in generated HTML just makes the template difficult to read. Not-so-great indentation in automatically generated HTML is fine. Keeping the template indented and readable on the other hand, is necessary.

File: tgarchive/example/template.html (hunk 4 above)
Comment (knadh) [reply tgarchive/example/template.html:25]: Users can easily remove this if they don't want it once the template it generated. I think it's okay for it to be there by default.

File: tgarchive/example/template.html (hunk 1 above)
Comment (knadh) [reply tgarchive/example/template.html:43]: In this context, it is always going to be an auto generated line of text. `<p>` is thus fine. If someone wants to add multiple paragraphs there by editing the template, they can always change the markup.

File: tgarchive/example/template.html [hunk 6]
```diff
@@ -39,163 +40,171 @@
...
-		        				<a href="{{ m.slug }}.html">
-		        					{{ m.label }}
-		        					<span class="count">({{ m.count }})
-		        				</span></a>
-		        			</li>
-		        		{% endfor %}
-		        	</ul>
-		        </li>
-		        {% endfor %}
+			<ul class="timeline index">{%
```
Comment (knadh) [inline tgarchive/example/template.html:49]: This is very difficult to read. See above comment about indentation being important in the raw template. It has no importance in the generated HTML.

**Comment by knadh** [comment]
Joining lines in the template file to remove indentation in generated HTML just makes the template very difficult to read. Not-so-great indentation in automatically generated HTML is fine. It has little importance.

Keeping the template indented and readable on the other hand, is necessary.

**Comment by milahu** [comment]
makes sense ... more user friendly

minimal solution:
remove empty indented lines = remove trailing whitespace
these are produced by the indented template tags `{% ... %}`
in `git diff` these show up as red blocks (not so pretty)

```py
bytes = re.sub(b" +\n", b"\n", bytes) # remove empty indented lines
```

**Comment by knadh** [comment]
`-%}` instead of `%}` strips off the extra new line in the end. You can amend the PR with this.

**Comment by knadh** [comment]
Thanks. However, this PR now has many unrelated changes. Per day count (and related flags), template generation options, updates to dependencies, new template variables, HTML fixes ... Very difficult to review, and revert if something goes wrong.

Could you please resend these as distinct PRs? The minor template fixes can be one PR.

**Comment by milahu** [comment]
i found these patches too trivial to make separate PRs

> Very difficult to review, and revert if something goes wrong.

[review by commit](https://github.com/knadh/tg-archive/pull/69/commits)?

merge with merge commit? (dont squash)

**Comment by knadh** [comment]
Apologies, I finally got a chance to pull this locally and while these are a bunch of minor changes, I'm unable to make sense of them. Minor template fixes can be one commit, but the new features in config, could you please separate the PRs?

**Comment by milahu** [comment]
nope sorry

**Comment by knadh** [comment]
Alright. I'll close this PR and try to copy bits that I'm able to figure out easily from it and merge it as I find time. Thanks.

**Pull request closed without merging**